docs/the-new-architecture/pillars-turbomodule.md

```diff
@@ -3,16 +3,699 @@ id: pillars-turbomodules
+
+First, to enable the **New Architecture**:
+
+1. Open the `android/gradle.properties` file
+2. Scroll down to the end of the file and switch the `newArchEnabled` property from `false` to `true`.
+
+Then, to manually link your new TurboModule:
+
+1. Open the `NewArchitecture/android/app/build.gradle` file and update the file as it follows:
+    ```diff
+        "PROJECT_BUILD_DIR=$buildDir",
+        "REACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
+    -   "REACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build"
+    +   "REACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build",
+    +   "NODE_MODULES_DIR=$rootDir/../node_modules/"
+        cFlags "-Wall", "-Werror", "-fexceptions", "-frtti", "-DWITH_INSPECTOR=1"
+        cppFlags "-std=c++17"
+    ```
+1. Open the `NewArchitecture/android/app/src/main/jni/Android.mk` file and update the file as it follows:
+    ```diff
+        # If you wish to add a custom TurboModule or Fabric component in your app you
+        # will have to include the following autogenerated makefile.
+        # include $(GENERATED_SRC_DIR)/codegen/jni/Android.mk
+    +
+    +   include $(NODE_MODULES_DIR)/rtn-calculator/android/build/generated/source/codegen/jni/Android.mk
+        include $(CLEAR_VARS)
+    ```
+1. In the same file above, go to the `LOCAL_SHARED_LIBRARIES` setting and add the following line:
+    ```diff
+        libreact_codegen_rncore \
+    +   libreact_codegen_RTNCalculator \
+        libreact_debug \
+    ```
+1. Open the `NewArchitecture/android/app/src/main/jni/MainApplicationModuleProvider.cpp` file and update the file as it follows:
+    1. Add the import for the calculator:
+        ```diff
+            #include <answersolver.h>
+        +   #include <RTNCalculator.h>
+        ```
+    1. Add the following check in the `MainApplicationModuleProvider` constructor:
+        ```diff
+            // auto module = samplelibrary_ModuleProvider(moduleName, params);
+            // if (module != nullptr) {
+            //    return module;
+            // }
+
+        +    auto module = RTNCalculator_ModuleProvider(moduleName, params);
+        +    if (module != nullptr) {
+        +        return module;
+        +    }
+
+            return rncore_ModuleProvider(moduleName, params);
+        }
+        ```
+
+### JavaScript
+
+Now you can use your TurboModule calculator in your app!
+
+Here's an example App.js file using the `add` method:
+
+```js title="App.js"
+/**
+ * Sample React Native App
+ * https://github.com/facebook/react-native
+ *
+ * @format
+ * @flow strict-local
+ */
+import React from 'react';
+import {useState} from 'react';
+import type {Node} from 'react';
+import {SafeAreaView, StatusBar, Text, Button} from 'react-native';
+import RTNCalculator from 'rtn-calculator/js/NativeCalculator.js';
+
+const App: () => Node = () => {
+  const [currentResult, setResult] = useState<number | null>(null);
```

Comment:
```suggestion
  const [result, setResult] = useState<number | null>(null);
```

Reply:
Why this change, just to make it match the `[x, setX]` pattern? not a big deal, just will need to rename the `const` in the `onPress` to avoid conflict

Reply:
updated

Reply:
> just to make it match the [x, setX] pattern?

Essentially yes. That's a nit, but the rest of docs we have that rely on `useState` follow this convention